commitlint/tests/tsconfig.json

```diff
@@ -0,0 +1,9 @@
+{
+    "compilerOptions": {
+        "strict": true,
+        "lib": ["ES2021"],
+        // TODO: explore replacing the below setting with just target:ES2015
+        "downlevelIteration": true,
+        "types": ["node", "jest"],
+    }
+}
```

Comment:
@webwarrior-ws isn't there a way to have a single tsconfig file for the whole repo? to respect DRY

Reply:
We need different settings for tests, as they use CommonJS modules and not ES modules.

Reply:
Ok, what would be the best way to fix this? Maybe have a 3rd commit in this PR that migrates the tests from jest to yarn? I only suggested the latter because it seems that it is `yarn test` what commitlint upstream uses to run their tests, but correct me if I'm wrong.

Reply:
AFAIK yarn is just a runner. commitlint uses `vitest`:
https://github.com/conventional-changelog/commitlint/blob/e41023d01c3f8569c3b91f4362df9bdc645b5f3f/package.json#L25

Reply:
Added a commit that migrates from jest to vitest